db/migrate/20210201055410_create_suspended_users.rb

```diff
@@ -0,0 +1,11 @@
+class CreateSuspendedUsers < ActiveRecord::Migration[6.0]
+  def change
+    create_table :suspended_users do |t|
+      t.string :username_hash, null: false
+
+      t.timestamps
+    end
+
+    add_index :suspended_users, :username_hash, unique: true
```

Comment:
This may be paranoid, but on the off-chance we do get a hash collision there's a uniqueness validation....